### ISSUE TITLE
transforms.Normalize(mean=[0.485, 0.456, 0.406],  std=[0.229, 0.224, 0.225])   ?

### ISSUE THREAD

**Issue by zhangquanwei962** [JarrentWu1031/CCPL#12]
Hi, thank you for your greate work! I find that your code was not normalized with the pre trained ImageNet value, as shown in the title.
So I add this to the code, but when I use the trained weight to test, the performance is not good.
![brad_pitt_stylized_in2](https://user-images.githubusercontent.com/61458769/205535860-19e187e4-ffac-4d54-b0fd-9677366cf8a2.jpg)
In contrast, I remove regularization and get the same result as yours.
![brad_pitt_stylized_in2](https://user-images.githubusercontent.com/61458769/205536212-de787ceb-cacb-46e8-8f07-e5c3bef2a240.jpg)
So why this difference？ Can you tell me the reason? thank you!

**Comment by JarrentWu1031** [comment]
> Hi, thank you for your greate work! I find that your code was not normalized with the pre trained ImageNet value, as shown in the title. So I add this to the code, but when I use the trained weight to test, the performance is not good. ![brad_pitt_stylized_in2](https://user-images.githubusercontent.com/61458769/205535860-19e187e4-ffac-4d54-b0fd-9677366cf8a2.jpg) In contrast, I remove regularization and get the same result as yours. ![brad_pitt_stylized_in2](https://user-images.githubusercontent.com/61458769/205536212-de787ceb-cacb-46e8-8f07-e5c3bef2a240.jpg) So why this difference？ Can you tell me the reason? thank you!

Hello, Zhang! Thanks for your appreciation. Our code is based on the pytorch implementation of the [**AdaIN**](https://github.com/naoto0804/pytorch-AdaIN) paper. As you can see in [**issue 13**](https://github.com/naoto0804/pytorch-AdaIN/issues/13), the original implementation does not involve input normalization, too. I guess it might not be necessary to pre-normalize the input images in stylization tasks. You could try to add pre-normalization during and see how it works in the inference stage.

**Comment by zhangquanwei962** [comment]
> > Hi, thank you for your greate work! I find that your code was not normalized with the pre trained ImageNet value, as shown in the title. So I add this to the code, but when I use the trained weight to test, the performance is not good. ![brad_pitt_stylized_in2](https://user-images.githubusercontent.com/61458769/205535860-19e187e4-ffac-4d54-b0fd-9677366cf8a2.jpg) In contrast, I remove regularization and get the same result as yours. ![brad_pitt_stylized_in2](https://user-images.githubusercontent.com/61458769/205536212-de787ceb-cacb-46e8-8f07-e5c3bef2a240.jpg) So why this difference？ Can you tell me the reason? thank you!
> 
> Hello, Zhang! Thanks for your appreciation. Our code is based on the pytorch implementation of the [**AdaIN**](https://github.com/naoto0804/pytorch-AdaIN) paper. As you can see in [**issue 13**](https://github.com/naoto0804/pytorch-AdaIN/issues/13), the original implementation does not involve input normalization, too. I guess it might not be necessary to pre-normalize the input images in stylization tasks. You could try to add pre-normalization during and see how it works in the inference stage.

Thank you for your reply. I guess that I have found why the difference occuried. 
`def denormalzation(tensor, device):
    mean = torch.tensor([0.485, 0.456, 0.406]).reshape(-1, 1, 1).to(device)
    std = torch.tensor([0.229, 0.224, 0.225]).reshape(-1, 1, 1).to(device)
    tensor = torch.clamp(tensor * std + mean, 0, 1)
    return tensor`
I made a mistake that I used the `torch.clamp(tensor * mean + std)`. Thank you again. It's really not necessary. I will verify the modified results.
